src/wg/color/rgb/format.d

```diff
@@ -47,9 +48,274 @@ struct RGBFormatDescriptor
 
     ushort componentsPresent;
 
-    const(ComponentDesc)[] components;
+    ComponentDesc[] components;
 
     const(RGBColorSpace)* colorSpace;
 
-    string userData;
+    const(char)[] userData;
+}
+
+/**
+ * Parse RGB format descriptor from string.
+ */
+RGBFormatDescriptor parseRGBFormat(const(char)[] format) @trusted pure
+{
+    import std.exception : enforce;
+
+    RGBFormatDescriptor r;
+
+    // parse data into stack buffers
+    RGBFormatDescriptor.ComponentDesc[32] components = void;
+    RGBColorSpace cs = void;
+    string error = format.parseRGBFormat(r, components, cs);
+    enforce(error == null, format ~ " : " ~ error);
```

Comment:
Note: parsing data on the stack, no GC allocations unless parsing is successful.